siliconcompiler/schema/schema_cfg.py

```diff
@@ -803,14 +836,15 @@ def schema_datasheet(cfg, design='default', name='default', mode='default'):
 
     # Pin type
     scparam(cfg, ['datasheet', design, 'pin', name, 'type', mode],
-            sctype='str',
+            sctype='enum',
+            enum=['digital', 'analog', 'clock', 'supply', 'ground'],
             shorthelp="Datasheet: pin type",
             switch="-datasheet_pin_type 'design name mode <str>'",
             example=[
-                "cli: -datasheet_pin_type 'mydevice vdd type power'",
-                "api: chip.set('datasheet','mydevice','pin','vdd','type','global','power')"],
+                "cli: -datasheet_pin_type 'mydevice vdd global supply'",
+                "api: chip.set('datasheet','mydevice','pin','vdd','type','global','supply')"],
             schelp="""Pin type specified on a per mode basis. Acceptable pin types
-            include: digital, analog, clk, power, ground""")
+            include: digital, analog, clock, supply, ground""")
```

Comment:
The list of enums is already printed in the docs, so it's not needed to add them in the help text as well.